download-utils/src/lib.rs

```diff
@@ -274,16 +274,19 @@ pub fn download_snapshot_archive<'a, 'b>(
         ArchiveFormat::TarBzip2,
         ArchiveFormat::Tar, // `solana-test-validator` creates uncompressed snapshots
     ] {
+        let snapshot_archives_remote_dir = snapshot_archives_dir.to_path_buf().join("remote");
+        fs::create_dir_all(snapshot_archives_remote_dir.to_path_buf()).unwrap();
```

Comment:
Will this `.unwrap()` assert if the directory already exists? If the archive format is not a TarZstd, then the directory would've already been created, right? Maybe this code needs to be hoisted out of the loop.

Or alternatively the `snapshot_archives_dir` parameter could already be the remote directory?

Reply:
https://doc.rust-lang.org/std/fs/fn.create_dir_all.html
Based on the documentation, it shouldn't assert. unlike `fs::create_dir`.
Yes, we should move the code out of the loop.